### PR TITLE
[Major] onChange and onBlur undefined error and date parameters in Date

### LINKED ISSUE #902: Change `onChange` and `onBlur` event in Date Input

### PR DESCRIPTION
**Pull request by marcialfps**
- onChange and onBlur undefined error and date parameters in Date input.
- Improved description of `error` prop.
- Avoid jumps and improvements in examples.

Closes #902

### CODE REVIEW

File: lib/src/date-input/types.ts [hunk 1]
```diff
@@ -57,14 +57,14 @@ type Props = {
    * If the string value is a valid date, error will
    * be null. Also, if the string value is not a valid date, date will be null.
```
Comment (Jialecl) [inline lib/src/date-input/types.ts:58]: The comment needs to be updated as the documentation

File: lib/src/date-input/types.ts [hunk 2]
```diff
@@ -57,14 +57,14 @@ type Props = {
...
    */
-  onChange?: (val: { value: string; error: string | null; date: Date | null }) => void;
+  onChange?: (val: { value: string; error?: string; date?: Date }) => void;
   /**
    * This function will be called when the input element loses the focus.
    * An object including the string value, the error and the date value
    * will be passed to this function. If the string value is a valid date, error will
    * be null. Also, if the string value is not a valid date, date will be null.
    */
```
Comment (Jialecl) [inline lib/src/date-input/types.ts:66]: The error portion of the comment needs to be updated as the documentation.

Comment (Jialecl) [reply lib/src/date-input/types.ts:66]: Same as the error property below.

File: lib/src/date-input/types.ts (hunk 1 above)
Comment (marcialfps) [reply lib/src/date-input/types.ts]: Done!

File: lib/src/date-input/types.ts (hunk 2 above)
Comment (marcialfps) [reply lib/src/date-input/types.ts:66]: 👍🏼

**Review by Jialecl — APPROVED** [review]

**Pull request merged**